Components/CategoryResolver/DefaultCategoryResolver.php

```diff
@@ -16,26 +16,48 @@ class DefaultCategoryResolver extends CategoryResolver
      */
     public function resolve(array $categories, $shopId, $stream)
     {
-        $remoteCategoriesIds = $this->manager->getConnection()->executeQuery('
-            SELECT id
-            FROM s_plugin_connect_categories
-            WHERE shop_id = ? AND category_key IN (?)',
-            [$shopId, array_keys($categories)],
-            [\PDO::PARAM_INT, \Doctrine\DBAL\Connection::PARAM_STR_ARRAY])->fetchAll(\PDO::FETCH_COLUMN);
-
-        return $this->manager->getConnection()->executeQuery('
-            SELECT local_category_id
-            FROM s_plugin_connect_categories_to_local_categories
-            WHERE remote_category_id IN (?) AND (stream = ? OR stream IS NULL) ',
-            [$remoteCategoriesIds, $stream],
-            [\Doctrine\DBAL\Connection::PARAM_INT_ARRAY, \PDO::PARAM_STR])->fetchAll(\PDO::FETCH_COLUMN);
+        $tree = $this->generateTree($categories);
+
+        return array_keys($this->traverseTree($tree, $shopId, $stream));
     }
 
     /**
-     * {@inheritdoc}
+     * @param array $children
+     * @param int $shopId
+     * @param string $stream
+     * @return array
      */
-    public function generateTree(array $categories, $idPrefix = '')
+    private function traverseTree(array $children, $shopId, $stream)
```

Comment:
how come the name change? traverse with what?

Reply:
generate Tree was an abstract function before 
the defaultCategoryResolver did just return an empty array in that function
that's why i extracted the logic from autoCategoryResolver to the base class